### PR TITLE
[BACKLOG-40760] Add doAction method for getVariable dialog

### PR DESCRIPTION
**Pull request by kdarshana12**

### CODE REVIEW

File: engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
  *
  * Pentaho Data Integration
  *
- * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
+ * Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
```
Comment (sVarma1992) [inline engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java:5]: change 2024 to 2002-2024

File: engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java [hunk 2]
```diff
@@ -38,9 +42,6 @@
 
 /**
  * Get information from the System or the supervising transformation.
- *
```
Comment (sVarma1992) [inline engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java:41]: why is this deleted we should not delete anything existing

File: engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java [hunk 3]
```diff
@@ -51,6 +52,7 @@ public GetVariable( StepMeta stepMeta, StepDataInterface stepDataInterface, int
     super( stepMeta, stepDataInterface, copyNr, transMeta, trans );
   }
 
+  @Override
```
Comment (sVarma1992) [inline engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java:55]: why is this added this is not our changes, same thing for the other two methods as well

File: engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java [hunk 4]
```diff
@@ -140,8 +143,40 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
     return false;
   }
 
+  @Override
   public void dispose( StepMetaInterface smi, StepDataInterface sdi ) {
     super.dispose( smi, sdi );
   }
 
+  public JSONObject doAction( String fieldName, StepMetaInterface stepMetaInterface, TransMeta transMeta,
```
Comment (sVarma1992) [inline engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java:151]: Map<String, String> queryParams 
The above param has been added to the doAction method
Please rebase with latest feature branch

File: engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java (hunk 3 above)
Comment (kdarshana12) [reply engine/src/main/java/org/pentaho/di/trans/steps/getvariable/GetVariable.java]: Added because of check-style suggestion. Removed in latest push.
